lib/logstash/filters/mutate.rb

```diff
@@ -428,12 +428,13 @@ def merge(event)
           next
         end
 
+        # No need to test the other
         if dest_field_value.is_a?(Hash)
-          # No need to test the other
-          event[dest_field].update(added_field_value)
+          # do not use event[dest_field].update to avoid assuming the in-place mutability of field values
```

Comment:
right, but not *immutable* strictly speaking, you can mutate it but it would be updated in Event itself...